core/src/main/scala/org/ensime/indexer/SearchService.scala

```diff
@@ -210,13 +215,17 @@ class SearchService(
   /** free-form search for classes */
   def searchClasses(query: String, max: Int): List[FqnSymbol] = {
     val fqns = index.searchClasses(query, max)
-    Await.result(db.find(fqns), QUERY_TIMEOUT) take max
+    (Await.result(db.find(fqns), QUERY_TIMEOUT) take max) sortBy { f =>
+      !applyBoost(f.file drop 7)
```

Comment:
yeah the DB should return hits in the same order that the indexer finds them. If you can prove that's not the case (with a unit test), I'll look at it and fix it (unless you get there first).